src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -472,13 +472,12 @@ InterpreterSelectQuery::InterpreterSelectQuery(
 
     /// Check support for FINAL for parallel replicas
     bool is_query_with_final = isQueryWithFinal(query_info);
-    if (is_query_with_final && (!settings.parallel_replicas_custom_key.value.empty() || settings.allow_experimental_parallel_reading_from_replicas > 0))
+    if (is_query_with_final && settings.allow_experimental_parallel_reading_from_replicas > 0)
```

Comment:
Was planning to do that but then I saw this comment
```
    /// TODO: Also disable parallel replicas in case of FINAL
```
I also cannot remove the comment because it's true for other type of parallel replicas